### ISSUE TITLE
Interior loading desync issue?

### ISSUE THREAD

**Issue by lynyx23** [IAmBatby/LethalLevelLoader#121]
MOR:
1. Create a 4 player lobby with MoreCompany. Have everyone connect.
2. As the host select a custom moon (I've set up the moons so that each one has it's own custom interior). Also flick the start lever
3. Enter the facility and observe some or all of the other players fall through the map
Example host pov: [medal.tv video](https://medal.tv/games/lethal-company/clips/ieqThI0w4kbOXJ3DL/d1337PTBFm9G?invite=cr-MSw5SzYsMjAwMDkxNDQ5LA)

Note:
My friends told me that while falling they can see that the map isn't rendered at all. The only thing that they see is a few random assets throughout the map and the spawned enemies. While spectating they still don't see any walls or terrain.

This is really frustrating. I'm playing on a 4 player lobby (I'm the host) with MoreCompany. I've seen that I'm not the only person on the internet encountering this type of stuff and I've tried to take some "precautions" (only the host selected the moon and flicked the start lever, connecting to the lobby at the same time, restarting the lobby after a "desync" issue).

If there is anything else I can submit, I'd be glad to do so.

**Comment by IAmBatby** [comment]
I'd need the player.log

**Comment by hghghghv** [comment]
> I'd need the player.log

Where to get it?

**Comment by hghghghv** [comment]
![image](https://github.com/IAmBatby/LethalLevelLoader/assets/109020456/788977ac-4a81-44ba-aee3-595be826fc31)

Can this help?

**Comment by lynyx23** [comment]
> > I'd need the player.log
> 
> Where to get it?

C:\Users\_<insert your user here>_\AppData\LocalLow\ZeekerssRBLX\Lethal Company

**Comment by lynyx23** [comment]
> I'd need the player.log

[host.log](https://github.com/user-attachments/files/16169158/host.log)
[friend.log](https://github.com/user-attachments/files/16169159/friend.log)

I repro'd this with 2 players (1 host 1 friend). My friend fell through the floor, but the level loading was different this time. Instead of not rendering walls and stuff, it loaded a vanilla mansion.

Here's our povs:
![friend_spectate](https://github.com/IAmBatby/LethalLevelLoader/assets/61161969/e4707737-22e1-4512-9b05-a531bfe2e0c0)
![host_pov](https://github.com/IAmBatby/LethalLevelLoader/assets/61161969/8b3e234b-8aea-4c83-ac5a-5c24dcdaed68)

**Comment by Phantom139** [comment]
This is the same desync issue I ran into with my friends where the host would generate one interior while other players would generate a different one and subsequently fall through the map. Looking at the log files you can see that the host / friend have different IDs for the Nyx moon.

What is happening is that the custom levels are loading in a different order for each player, resulting in the ExtendedCustomLevels list having different identifiers for each player. I pushed a PR (#112) a few weeks back with a small bit of code to sort that list by name after all of the levels have been loaded in which seemed to address this issue when playing a few test rounds w/ friends. Obviously this code would need to be updated to V55, but it should fix this problem.

**Comment by lynyx23** [comment]
Thanks for the reply Phantom139! Well I hope @IAmBatby will take your fix into consideration or implement something else that fixes the id desync issue. Thanks for the support and I hope this will be fixed!

**Comment by ZigzagAwaka** [comment]
Well, idk if it's the same problem as #109 but when you set up custom interiors for moons you have to enable custom content configuration in the config file
And bc of a bug if you enable this setting for a moon, every custom config will be ignored (no custom dungeons, no custom scraps, no custom ennemies), I still don't know if this a host or a client issue, but in any case this prevent from playing custom content with LLL

**Comment by danielbro9** [comment]
> Well, idk if it's the same problem as #109 but when you set up custom interiors for moons you have to enable custom content configuration in the config file And bc of a bug if you enable this setting for a moon, every custom config will be ignored (no custom dungeons, no custom scraps, no custom ennemies), I still don't know if this a host or a client issue, but in any case this prevent from playing custom content with LLL

When you set up custom interiors for moons you only have to enable the custom content configuration for the interior itself, not the moons you tell it to generate on.

**Comment by ZigzagAwaka** [comment]
> > Well, idk if it's the same problem as #109 but when you set up custom interiors for moons you have to enable custom content configuration in the config file And bc of a bug if you enable this setting for a moon, every custom config will be ignored (no custom dungeons, no custom scraps, no custom ennemies), I still don't know if this a host or a client issue, but in any case this prevent from playing custom content with LLL
> 
> When you set up custom interiors for moons you only have to enable the custom content configuration for the interior itself, not the moons you tell it to generate on.

Yes, but then if you want to make Old Bird spawn on moons that are not updated to v50+ you have to enable custom content config for that said moon
Secret Labs can't spawn Old Bird naturally but you can add them with LLL config, but then it breaks every custom content (dungeons, some items, mods etc)

**Comment by Phantom139** [comment]
I had a recent playtest session with my friend (~3 hours), and we only encountered one issue of falling through the map, although both friends who fell through said it was a hole in the map they fell into and not an interior desync.

For now, I would day that this is "probably" fixed, but I would still recommend additional testing to be sure.
